### PR TITLE
fix(router): fix missing filter part for available endpoints

### PR DESCRIPTION
**Pull request by jwx**
# Pull Request

## 📖 Description

This PR adds a filter that was overlooked in the navigation coordinator refactor, fixing an issue mentioned in Discord that could occur when using default child routes.

## 📑 Test Plan

- Make sure existing tests pass
- Add test for the specific occasion

### CODE REVIEW

File: packages/__tests__/src/router/router.link-click-defaults.spec.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    const test = tests[i];
+    it(`for "${test.load}"`, async function () {
+      const { platform, host, router, $teardown } = await $setup({}, [Nav, Parent, Child, GrandChild]);
+
+      await $load('/nav', router, platform);
+      await platform.domQueue.yield();
+
+      const links = host.getElementsByTagName('A') as unknown as HTMLElement[];
+      const link = links[i];
+      link.click();
```
Comment (bigopon) [inline packages/__tests__/src/router/router.link-click-defaults.spec.ts:152]: these dom targeting will be confusing, and prone to breaking during refactoring. Is there any better way to express this?

File: packages/__tests__/src/router/router.link-click-defaults.spec.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    ];
+  });
+
+  const tests = [
+    { load: '/parent/child', result: '!parent!!child!!grandchild!', },
+  ];
+
+  for (let i = 0; i < tests.length; i++) {
+    const test = tests[i];
+    it(`for "${test.load}"`, async function () {
```
Comment (bigopon) [inline packages/__tests__/src/router/router.link-click-defaults.spec.ts:144]: is there a better name for this test?

**Review by bigopon — APPROVED** [review]
Looks good @jwx , just some small comments for clarity

File: packages/__tests__/src/router/router.link-click-defaults.spec.ts (hunk 2 above)
Comment (jwx) [reply packages/__tests__/src/router/router.link-click-defaults.spec.ts]: Yes. Updated.

**Pull request merged**